src/PRDevice.cpp

```diff
@@ -1088,11 +1088,12 @@ PRResult PRDevice::VerifyChipID()
                 buffer[i] = requestedDataQueue.front();
                 requestedDataQueue.pop(); // Ignore address word.  TODO: Verify the address.
             }
-            if (buffer[1] != P_ROC_CHIP_ID && buffer[1] != P3_ROC_CHIP_ID) 
+            if (buffer[1] != P_ROC_CHIP_ID && buffer[1] != P3_ROC_CHIP_ID)
             {
                 DEBUG(PRLog(kPRLogError, "Error in VerifyID(): Dumping buffer\n"));
-                for (i = 0; i < bufferWords; i++) 
+                for (i = 0; i < bufferWords; i++)
                     DEBUG(PRLog(kPRLogError, "buffer[%d]: 0x%x\n", i, buffer[i]));
+                PRSetLastErrorText("Chip ID does not match.");
                 rc = kPRFailure;
```

Comment:
Some additional `PRSetLastErrorText()` calls here and below.

Reply:
This was to find the flasher error. I then added an error to every error path so that empty error no longer happen (hopefully).